### PR TITLE
Support more `JSON_TABLE` functionality

### PR DESCRIPTION
**Pull request by JCOR11599**
Source: https://dev.mysql.com/doc/refman/8.0/en/json-table-functions.html
```
JSON_TABLE(
    expr,
    path COLUMNS (column_list)
)   [AS] alias

column_list:
    column[, column][, ...]

column:
    name FOR ORDINALITY
    |  name type PATH string path [on_empty] [on_error]
    |  name type EXISTS PATH string path
    |  NESTED [PATH] path COLUMNS (column_list)

on_empty:
    {NULL | DEFAULT json_string | ERROR} ON EMPTY

on_error:
    {NULL | DEFAULT json_string | ERROR} ON ERROR
```

Note: the MySQL docs indicate that `PATH` is optional in the `NESTED` case, but it doesn't seem that way.
I chose to follow what they say rather than what they do.

### CODE REVIEW

File: go/vt/sqlparser/parse_test.go [hunk 1]
```diff
@@ -7268,11 +7271,122 @@ FROM
...
+				")) as jt",
+		},
+		{
+			input: `SELECT * FROM JSON_TABLE('{}', '$' COLUMNS( c1 INT PATH '$.c1' ERROR ON ERROR )) as jt;`,
+			output: "select * from json_table('{}', \"$\" columns(\n" +
+				"\tc1 INT path \"$.c1\" error on error\n" +
+				")) as jt",
+		},
+
+		// NESTED PATH TESTS
```
Comment (fulghum) [inline go/vt/sqlparser/parse_test.go:7364]: Would be nice to test a double `NESTED` statement.

**Review by fulghum — APPROVED** [review]
Nice work! Just a few small notes. The JSON table types and yacc rules look cleaner than before. Good job!

File: go/vt/sqlparser/token.go [hunk 2]
```diff
@@ -462,6 +462,7 @@ var keywords = map[string]int{
 	"names":                         NAMES,
 	"national":                      NATIONAL,
 	"natural":                       NATURAL,
+	"nested":                        NESTED,
```
[the remark this answers is not in the document]
Comment (JCOR11599) [reply go/vt/sqlparser/token.go:465]: `NESTED` is already in `non_reserved_keywords`

File: go/vt/sqlparser/ast.go [hunk 3]
```diff
@@ -2424,6 +2424,8 @@ type TableSpec struct {
 	Indexes     []*IndexDefinition
 	Constraints []*ConstraintDefinition
 	Options     string
+	Nested      bool
+	Path        string
```
[the remark this answers is not in the document]
Comment (JCOR11599) [reply go/vt/sqlparser/ast.go:2428]: `Path` is used in `Format()` and in GMS, but `Nested` isn't; will remove.

[the remark this answers is not in the document]
Comment (JCOR11599) [reply go/vt/sqlparser/ast.go:2428]: Just realized this is for `TableSpec` and not for `JSONTableSpec`, will remove unused member vars from both.

**Pull request merged**